File: app/src/main/java/ntutifm/game/google/net/RetrofitManager.java
package ntutifm.game.google.net;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitManager {
    // 以Singleton模式建立
    private static RetrofitManager mInstance = new RetrofitManager();
    private MyAPIService myAPIService;
    private RetrofitManager() {

        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl("http://192.168.43.178:80")
                .addConverterFactory(GsonConverterFactory.create())
                .build();

        myAPIService = retrofit.create(MyAPIService.class);
    }

    public static RetrofitManager getInstance() {
        return mInstance;
    }

    public MyAPIService getAPI() {
        return myAPIService;
    }
}

File: app/src/main/java/ntutifm/game/google/net/RetrofitManager.kt
package ntutifm.game.google.net

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class RetrofitManager {
    companion object {
        private var instance: RetrofitManager? = null
        private var _api: MyAPIService? = null
        val api get() = _api
        init{
            if (instance == null) {
                instance = RetrofitManager()
                val retrofit = Retrofit.Builder()
                    .baseUrl("https://ntutifm.zeabur.app/")
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()
                _api = retrofit.create(MyAPIService::class.java)
            }
        }
    }
}